### ISSUE TITLE
connect function does not work when inherited

### ISSUE THREAD

**Issue by HilkoLantinga** [alexkwolfe/node-eversocket#4]
```JavaScript
var util = require('util'), EverSocket = require('eversocket').EverSocket;
var MySocket = function(config) {'use strict';
    EverSocket.call(this,{
        reconnectWait : 100,
        timeout : 100,
        reconnectOnTimeout : true
    });
};
util.inherits(MySocket, EverSocket);
var socket = new MySocket();
socket.connect(4999);
```
/node_modules/eversocket/index.js:100
args.forEach(function handle(arg) {
       ^
RangeError: Maximum call stack size exceeded
________________________________________________________________________
It also does not support all net.connect() options, like an object as argument or localAddress

**Comment by brndmg** [comment]
Fixed with PR: https://github.com/alexkwolfe/node-eversocket/pull/5

**Comment by alexkwolfe** [comment]
Fixed in 7bef65fc147ef758e382e1d88fb8e693b72ab973. Thanks for the PR.
